src/main/java/by/custom/utilcalculator/domain/UserProgress.java

```diff
@@ -8,50 +8,118 @@
 import java.util.Map;
 
 public class UserProgress implements Serializable {
-    private TransportType transportType = null;
+    private GeneralTransportType generalTransportType = null;
     private CountryOrigin countryOrigin = null;
     private OwnersType ownersType = null;
     private CarAge carAge = null;
-    private TypeOfEngine typeOfEngine = null;
-    private VolumeOfEngine volumeOfEngine = null;
+    private M1TypeOfEngine typeOfEngine = null;
+    private M1EngineVolume m1EngineVolume = null;
+    private ExceptM1TransportType exceptM1TransportType = null;
+    private N1N3TransportWeight transportWeightN1N2N3 = null;
+    private M2M3EngineType engineTypeM2M3 = null;
+    private M2EngineVolume m2EngineVolume = null;
+    private TruckUnitClass truckUnitClass = null;
+    private TruckUnitWeight truckUnitWeight = null;
+    private TrailerO4Type trailersO4Type = null;
     private final String chatID;
-    private Step currentQuestion;
+    private Step nextStep;
 
     public UserProgress(final String chatID) {
         this.chatID = chatID;
     }
 
-    public TransportType getTransportType(){
-        return transportType;
+    public void setNextStep(final Step nextStep) {
+        this.nextStep = nextStep;
+    }
+
+    public Step getNextStep(){
+        return nextStep;
+    }
+
+    public GeneralTransportType getGeneralTransportType() {
+        return generalTransportType;
     }
 
     public CountryOrigin getCountryOrigin() {
         return countryOrigin;
     }
 
     public Command[] getUserPath() {
```

Comment:
Already forgot how we use the user path. Please, remember me to discuss it on our call. Now it looks strange.

Reply:
Too many steps. Possible we have to fill it after each user command and not calculate it. We will discuss it.

Reply:
I use getUserPath() twice.
1. For getting the current user's node. Class CommandTree - method getNode(final UserProgress userProgress) returns current user's node. 
2. For debugging to check this array filling - what commands there are.